libnetwork/drivers/bridge/setup_ip_tables_linux.go

```
@@ -246,7 +246,7 @@ func (n *bridgeNetwork) setupIPTables(ipVersion iptables.IPVersion, maskedAddr *
 				"ipset":  ipsetName,
 				"bridge": config.BridgeName,
 				"subnet": maskedAddr,
-			}).Warnf("Subnet was already in the ipset")
+			}).Debug("Subnet was already in the ipset")
```

Perhaps in a follow-up we could add a comment inside this if-branch that this is an expected situation (as you mentioned in the PR description), and not another case of "libnetwork just out of convenience ignoring errors")

Would flipping the condition make that more explicit? (not sure 🤔 but happy to hear if you think so);

```go
if errors.Is(err, nl.IPSetError(nl.IPSET_ERR_EXIST)) {
	// re-adding an IP address to an ipset on firewalld reload is expected, and not an error.
	log.G(context.TODO()).WithFields(log.Fields{
		"ipset":  ipsetName,
		"bridge": config.BridgeName,
		"subnet": maskedAddr,
	}).Debug("Subnet was already in the ipset")
}
return fmt.Errorf("failed to add bridge %s (%s) to ipset: %w", config.BridgeName, maskedAddr, err)
```

---

ARF; ignore my suggestion of flipping the condition, as we should not return, so that's a bad suggestion.

(Adding a comment would still be good though 😅)

---

Thank you - comment added (and rebased).